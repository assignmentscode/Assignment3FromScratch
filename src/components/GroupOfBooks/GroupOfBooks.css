.same-author {
    border : 5px solid orange;
    border-radius: 50px;
    display: flex;
    justify-self: center;
}
.content-width {
    max-width: 80%;
    margin-left: auto;
    margin-right: auto;
}
.book-cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    margin: auto;
}
div span {
    background-color: white;
    color: orange;
    width: fit-content;
    text-align: center;
    /* margin-left: 44.5%; */
    font-size: 1.8em;
}
.author-name {
    display: flex;
    margin-bottom: -1.5%;
    justify-content: center;
}