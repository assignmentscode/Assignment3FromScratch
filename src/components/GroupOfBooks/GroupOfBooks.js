import React from 'react';
import './GroupOfBooks.css';
import BookCard from '../BookCard/BookCard';

const GroupOfBooks = () => (
  <div className="content-width">
    <div className="author-name">
      <span>J K Rowling</span>
    </div>
    <div className="same-author">
      <div className="book-cards">
        <BookCard Name="Harry" rating="4.2" likedData />
        <BookCard Name="Harry" rating="4.2" likedData />
        <BookCard Name="Harry" rating="4.2" likedData />
        <BookCard Name="Harry" rating="4.2" likedData />
      </div>
    </div>
  </div>
);
export default GroupOfBooks;
