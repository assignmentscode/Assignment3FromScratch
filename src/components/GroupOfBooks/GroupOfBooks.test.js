import React from 'react';
import renderer from 'react-test-renderer';
import GroupOfBooks from './GroupOfBooks';

describe('Button', () => {
  it('should render a div containing books of same author', () => {
    const tree = renderer.create(<GroupOfBooks />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
