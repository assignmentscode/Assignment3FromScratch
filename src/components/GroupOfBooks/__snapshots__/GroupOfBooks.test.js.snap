// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a div containing books of same author 1`] = `
<div
  className="content-width"
>
  <div
    className="author-name"
  >
    <span>
      J K Rowling
    </span>
  </div>
  <div
    className="same-author"
  >
    <div
      className="book-cards"
    >
      <div
        className="card"
      >
        <img
          alt="Book"
          className="blog-image"
          src="cover_image.jpeg"
        />
        <p
          className="post-heading"
        >
          Harry
        </p>
        <div>
          <span
            className="ratings "
          >
            4.2
          </span>
          <img
            alt="Heart"
            className="heart-logo"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
      <div
        className="card"
      >
        <img
          alt="Book"
          className="blog-image"
          src="cover_image.jpeg"
        />
        <p
          className="post-heading"
        >
          Harry
        </p>
        <div>
          <span
            className="ratings "
          >
            4.2
          </span>
          <img
            alt="Heart"
            className="heart-logo"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
      <div
        className="card"
      >
        <img
          alt="Book"
          className="blog-image"
          src="cover_image.jpeg"
        />
        <p
          className="post-heading"
        >
          Harry
        </p>
        <div>
          <span
            className="ratings "
          >
            4.2
          </span>
          <img
            alt="Heart"
            className="heart-logo"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
      <div
        className="card"
      >
        <img
          alt="Book"
          className="blog-image"
          src="cover_image.jpeg"
        />
        <p
          className="post-heading"
        >
          Harry
        </p>
        <div>
          <span
            className="ratings "
          >
            4.2
          </span>
          <img
            alt="Heart"
            className="heart-logo"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
